src/persist-client/src/internal/state.rs

```diff
@@ -1194,13 +1195,19 @@ where
         let all_empty_reqs = merge_reqs
             .iter()
             .all(|req| req.inputs.iter().all(|b| b.batch.is_empty()));
-        if record_compactions && claim_unclaimed_compactions && all_empty_reqs {
+        if record_compactions && all_empty_reqs {
+            let mut reqs_to_take = claim_compaction_percent / 100;
+            if (usize::cast_from(idempotency_token.hashed()) % 100)
+                < (claim_compaction_percent % 100)
+            {
+                reqs_to_take += 1;
+            }
             let threshold_ms = heartbeat_timestamp_ms.saturating_sub(lease_duration_ms);
             let min_writer = WriterKey::for_version(&crate::iter::MINIMUM_CONSOLIDATED_VERSION);
             merge_reqs.extend(
                 self.trace
                     .fueled_merge_reqs_before_ms(threshold_ms, Some(min_writer))
-                    .take(1),
+                    .take(reqs_to_take),
```

Comment:
Do we always want to be taking the first ones? Which order do they come out in (i.e. big->small, small->big, neither)?

No strong feels here if you're confident this is a good behavior, but let's add a comment with the answer to my above Q about the order.

Reply:
Pushed a comment - let me know if that reasoning doesn't sound right!